room/compiler/src/main/kotlin/androidx/room/solver/TypeAdapterStore.kt

```diff
@@ -532,9 +535,13 @@ class TypeAdapterStore private constructor(
                 rowAdapterLogs?.writeTo(context)
                 return rowAdapter
             }
-            if (query.runtimeQueryPlaceholder) {
-                // just go w/ pojo and hope for the best. this happens for @RawQuery where we
-                // try to guess user's intention and hope that their query fits the result.
+
+            // use pojo adapter as a last resort.
+            // this happens when @RawQuery or @SkipVerification is used.
+            if (typeMirror.isNotVoid() &&
```

Comment:
It might be redundant, but just to make it clear can you please also add a `query.resultInfo == null `, this should also fix the need to change `query.resultInfo == null`, and just to make it clear, we only want to try this pojo guessing if verification is off to make sure user opt-in to this.